modules/internal/localeModels/numa/LocaleModel.chpl

```diff
@@ -502,40 +496,16 @@ module LocaleModel {
     //
     const dnode =  chpl_nodeFromLocaleID(loc);
     const dsubloc =  chpl_sublocFromLocaleID(loc);
```

Comment:
Right, it's setSubloc that's slow (because it causes a shepherd migrate call in qthreads). Especially once we can do all the loc-id manipulations in (inlined) Chapel code (which we can't do yet because of limitations of the record support), getting the node and subloc parts here should be plenty fast.